File: entsoe-client/src/main/java/org/example/model/load/ObjectAggregation.java
package org.example.model.load;

import java.util.Arrays;
import lombok.AllArgsConstructor;
import lombok.Getter;

// Additional enum for Object Aggregation
@Getter
@AllArgsConstructor
public enum ObjectAggregation {
  AGGREGATED("A01", "Aggregated"),
  INDIVIDUAL("A02", "Individual");

  private final String code;
  private final String description;

  public static ObjectAggregation fromCode(String code) {
    return Arrays.stream(values())
        .filter(type -> type.code.equals(code))
        .findFirst()
        .orElseThrow(() -> new IllegalArgumentException("Unknown ObjectAggregation code: " + code));
  }
}

File: entsoe_client/src/entsoe_client/model/load/object_aggregation.py
from enum import Enum
from typing import Self

from entsoe_client.exceptions.unknown_object_aggregation_error import (
    UnknownObjectAggregationError,
)


class ObjectAggregation(Enum):
    AGGREGATED = ("A01", "Aggregated")
    INDIVIDUAL = ("A02", "Individual")

    def __init__(self, code: str, description: str) -> None:
        self.code = code
        self.description = description

    @classmethod
    def from_code(cls, code: str) -> Self:
        for member in cls:
            if member.code == code:
                return member
        raise UnknownObjectAggregationError(code)
